model/export_models.py

```diff
@@ -1,29 +1,63 @@
 import os
-import sys
+import subprocess
 
 import numpy as np
 import openvino as ov
 import requests
+import timm
 import torch
 from PIL import Image
 from PytorchWildlife.data import transforms as pw_trans
 from PytorchWildlife.models import detection as pw_detection
 from torchvision.transforms import InterpolationMode, transforms
-
-from domain.classify_detections import Model
+from tqdm import tqdm
 
 CROP_SIZE = 182
-BACKBONE = "vit_large_patch14_dinov2.lvd142m"
+WEIGHT_PATH = "deepfaune-vit_large_patch14_dinov2.lvd142m.pt"
 # Line too long
 URL = (
     "https://www.provincia.bz.it/agricoltura-foreste/"
     "fauna-caccia-pesca/images/braunbaer_6016_L.jpg"
 )
-WEIGHT_PATH = f"{os.path.join('../', 'deepfaune-vit_large_patch14_dinov2.lvd142m.pt')}"
 
-_thisdir = os.path.dirname(os.path.abspath(__file__))
-print(_thisdir)
-sys.path.append(os.path.join(_thisdir, "../src"))
+
+class Model(torch.nn.Module):
+    def __init__(self, weight_path, device="cpu", version="v1.1"):
+        """
+        Constructor of model classifier
+        """
+        super().__init__()
+
+        if not os.path.exists(weight_path):
+            self.download_weights(weight_path, version)
+
+        params = torch.load(weight_path, map_location=device, weights_only=False)
+        self.base_model = timm.create_model(
+            params["args"]["backbone"],
+            pretrained=False,
+            num_classes=params["args"]["num_classes"],
+            dynamic_img_size=True,
+        )
+        self.load_state_dict(params["state_dict"])
+
+    def forward(self, input):
+        return self.base_model(input)
+
+    def download_weights(self, fname, version):
+        url = f"https://pbil.univ-lyon1.fr/software/download/deepfaune/{version}/{fname}"
+        print(f"Downloading weights from {url}")

```

Comment:
Don't use f-strings with print()

Reply:
???

Reply:
print("Downloading weights from", url)

Reply:
Yeah I got that, but I do not understand why you suggest that? I get that for logger there is a (very negligible) performance boost as you do not do string concatenation if logger level is not high enough, but I do not understand what is the rationale here

Reply:
It's not about performance issues, rather about consistency with the behaviour of print(), which already converts all arguments in strings and separate them with spaces.

Reply:
Please have a look at the [python official i/o documentation](https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals). It doesn't seems to me that f-string are disallowed, also I never found a PEP guideline that explicitly discourage their use in `print` statements.

Again, if we want to enforce style rules we add them as a pre-commit check instead of having these discussions in PRs. However, I haven't found any of them about this specific case

Reply:
Yes, it wasn't in the official documentation / PEP8. This coding style is what I used/use on our code base at BMW, so I was automatically considering it.
We can avoid such discussions, np.

Reply:
np, I was just curious where this came from, like if there are some official PEP rule about. For instance, I didn't know the logging+f-string one before